easybuild/easyblocks/c/cfdemcoupling.py

```diff
@@ -0,0 +1,159 @@
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing CFDEMcoupling, implemented as an easyblock
+
+@author: Kenneth Hoste (Ghent University)
+"""
+import glob
+import os
+
+import easybuild.tools.environment as env
+import easybuild.tools.toolchain as toolchain
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import copy_file, mkdir, move_file
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.run import run_cmd
+
+
+class EB_CFDEMcoupling(EasyBlock):
+    """Support for building/installing CFDEMcoupling."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialisation of custom class variables for CFDEMcoupling."""
+        super(EB_CFDEMcoupling, self).__init__(*args, **kwargs)
+
+        self.build_in_installdir = True
+
+        self.cfdem_project_dir = os.path.join(self.installdir, '%s-%s' % (self.name, self.version))
+        self.liggghts_dir = os.path.join(self.installdir, 'LIGGGHTS-%s' % self.version)
+
+    def configure_step(self):
+        """Set up environment for building/installing CFDEMcoupling."""
+
+        # rename top-level directory to CFDEMcoupling-<version>
+        top_dirs = os.listdir(self.builddir)
+
+        for (pkgname, target_dir) in [('CFDEMcoupling', self.cfdem_project_dir), ('LIGGGHTS', self.liggghts_dir)]:
+            pkg_topdirs = [d for d in top_dirs if d.startswith(pkgname)]
+            if len(pkg_topdirs) == 1:
+                orig_dir = os.path.join(self.builddir, pkg_topdirs[0])
+                move_file(orig_dir, target_dir)
+            else:
+                raise EasyBuildError("Failed to find subdirectory for %s in %s %s", pkgname, self.builddir, top_dirs)
+
+        # always use env.setvar instead of os.putenv or os.environ for defining environment variables
+        env.setvar('CFDEM_VERSION', self.version)
+        env.setvar('CFDEM_PROJECT_DIR', self.cfdem_project_dir)
+
+        # define $CFDEM_PROJECT_USER_DIR to an empty existing directory
+        env.setvar('CFDEM_PROJECT_USER_DIR', os.path.join(self.builddir, 'CFDEM_PROJECT_USER_DIR'))
+        mkdir(os.getenv('CFDEM_PROJECT_USER_DIR'), parents=True)
```

Comment:
Seems odd to set a variable and read it right after. Not that it matters a lot, but it is certainly not efficient. Doesn't it make more sense to define a python variable and use it in both places?